src/material/tabs/tab-list.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {FocusOrigin} from '@angular/cdk/a11y';
+import {BooleanInput, NumberInput} from '@angular/cdk/coercion';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  ContentChildren,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  QueryList,
+  ViewChild,
+  ViewEncapsulation
+} from '@angular/core';
+
+import {
+  CanColor,
+  CanDisableRipple,
+  mixinColor,
+  mixinDisabled,
+  mixinDisableRipple
+} from '@angular/material/core';
+
+import {MAT_TAB_LABEL, MatTabLabel} from './tab-label';
+
+// Boilerplate for applying mixins to MatTabListLabel.
+/** @docs-private */
+const _MatTabListLabelMixinBase = mixinDisabled(class {});
+
+/** Directive for passing tab labels to tab list. */
+@Directive({
+  selector: 'mat-tab-list-label, matTabListLabel',
+  inputs: ['disabled'],
+})
+export class MatTabListLabel extends _MatTabListLabelMixinBase {
+  /** Aria label for the tab label. */
+  @Input() ariaLabel: string;
+
+  /**
+   * Reference to the element that the tab label is labelled by.
+   */
+  @Input() ariaLabelledby: string;
+
+  // TODO: Deduplicate with tab
+  /** Content for the tab label given by `<ng-template mat-tab-label>`. */
+  @ContentChild(MAT_TAB_LABEL)
+  get templateLabel(): MatTabLabel {
+    return this._templateLabelInput || this._templateLabel;
+  }
+  set templateLabel(value: MatTabLabel) {
+    this._setTemplateLabelInput(value);
+  }
+  protected _templateLabel: MatTabLabel;
+
+  /** Content for tab label passed as input. */
+  @Input('templateLabel') _templateLabelInput?: MatTabLabel;
```

Comment:
This will allow us to pass `templateLabel` as either input (when used with tab group) or as content (when tablist is consumed directly). Because the Tab#templateLabel is a CDKPortal, I couldn't find a clean way within tab group to pass the template to the tab list component as content.